clustered/common/src/main/java/org/ehcache/clustered/common/store/Link.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright Terracotta, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.ehcache.clustered.common.store;
+
+import java.io.Serializable;
+
+/**
+ * Building blocks of {@link Chain}
+ */
+public interface Link extends Serializable {
+
+  /**
+   * The links are associated with a sequence number
+   * generated by the server.
+   * This number is unique in a Chain =>
+   *  Same sequence number cannot be allotted to more than one link
+   *  in same chain.
+   *
+   * It is always an ever increasing positive number.
+   *
+   * @return the sequence number associated with this {@link Link}
+   */
+  long getSequenceNumber();
```

Comment:
The sequence numbers shouldn't need to be exposed through this interface.  The fact that sequence numbers even exist is just an implementation detail that we will use to optimize the replace operation.